website/docs/developer/research/research-deep-glacier-migration.md

```diff
@@ -0,0 +1,174 @@
+---
+id: research-deep-glacier-migration
+title: Deep archive storage migration Research Notes
+description: Research Notes on deep archive migration.
+---
+
+## Overview
+
+The goal of this wikipage is to provide an initial design and recommendation for migrating a customer's ORCA Flexible Retrieval (formerly Glacier) archive to DEEP ARCHIVE storage class. Deep archive S3 storage type is the cheapest glacier storage type currently present in AWS S3. For archive data that does not require immediate access such as backup or disaster recovery use cases, choosing `Deep Archive Access` tier could be a wise choice.
+- Storage cost is ~$1/TB data per month. 
+- Objects in the Deep Archive Access tier can be moved to the Frequent Access tier within 48 hours. Once the object is in the Frequent Access tier, users can send a GET request to retrieve the object. Like Flexible retrieval, the restored file is only available in frequent access for a configured number of days. For more information, see this [documentation](See https://docs.amazonaws.cn/en_us/AmazonS3/latest/userguide/restoring-objects-retrieval-options.html
+).
+
+### Assumptions
+
+- List of objects already exists in orca catalog and S3 bucket.
+- Users should be able to provide some information(filename) to retrieve objects possibly at the collection level.
+
+### Initial Implementation Idea
+
+A user might want to 
+- convert all the holdings or holdings with a known key prefix to DEEP ARCHIVE.
+- convert specific collection or granules to DEEP ARCHIVE which might not have the same prefix.
+
+#### Implementation idea for migrating all ORCA holdings to deep archive.
+
+- Use S3 lifecycle rule in this case possibly with a prefix option to migrate from glacier to deep archive.
+The following rule will move all objects with under prefix `tmp/` to DEEP ARCHIVE.
+
+```teraform
+resource "aws_s3_bucket_lifecycle_configuration" "bucket-config" {
+  bucket = aws_s3_bucket.bucket.bucket
+
+  rule {
+    id = "transition to deep archive"
+    status = "Enabled"
+
+    transition {
+      days          = 0
+      storage_class = "DEEP_ARCHIVE"
+    }
+    filter {
+      prefix = "tmp/"
+    }
+  }
+
+```
+:::tips
+To migrate to a different storage class, change `storage_class` under transition block to the desired class in the terraform code above.
+:::
+
+#### Implementation idea for migrating specific ORCA holdings to deep archive.
+
+S3 lifecycle rule will not work in this case. 
+Migration should be a 2 step process- first retrieve the objects and then copy the objects to deep archive.
+
+- Retrieve all objects currently stored with Glacier Flexible Retrieval type using `Bulk Retrieval` which is the cheapest retrieval option. Bulk retrievals are typically completed within 5–12 hours. The prices are lowest among all retrieval rates - $0.025 per 1,000 requests.
```

Comment:
You should add more information here:

How is this initiated? How do we know what objects to get? Where are we getting the file information from? Is there a limit to how much we can add to a bulk request (would we use batch or fan out for lambdas for performance)? Do we need to create a job in database to track status? How are we going to account for errors? Will end users be able to monitor the migration or get a status of % complete?

Reply:
there should be an optional terraform variable that user will input while deploying. 
The input should be a list of granuleId/collectionId- if true, it will run a lambda to restore and migrate those relevant files.

yes that is a possibility. Created [ORCA-499](https://bugs.earthdata.nasa.gov/browse/ORCA-499)  for implementing tracking. Users should be able to monitor the migration .Recommend looking into that in the card created above.